ddtrace/profiling/event.py

```diff
@@ -48,25 +48,15 @@ class StackBasedEvent(SampleEvent):
     trace_id = attr.ib(default=None, type=typing.Optional[int])
     span_id = attr.ib(default=None, type=typing.Optional[int])
     trace_type = attr.ib(default=None, type=typing.Optional[str])
-    trace_resource = attr.ib(default=None, type=typing.Optional[str])
+    trace_resource_container = attr.ib(default=None)
 
     def set_trace_info(
         self,
         span,  # type: typing.Optional[ddspan.Span]
     ):
-        # type: (...) -> None
-        if span is not None:
+        if span:
             self.trace_id = span.trace_id
             self.span_id = span.span_id
             if span._local_root is not None:
-                self.trace_resource = span._local_root.resource
                 self.trace_type = span._local_root.span_type
```

Comment:
small optimization we can make... use `span._local_root._span_type`, might help reduce the overhead of this code and total property getter calls ?

this could be made in this PR, but not required, just calling it out.

Reply:
fixed